File: frontend/src/utils/productAnalytics.ts
/**
 * Product Analytics Utility
 * 
 * Tracks user behavior within the Wordsworth AI platform.
 * Session-based tracking with localStorage persistence.
 */

import { getBackendUrl } from '@/config/api';

/**
 * Get or create session ID for analytics tracking.
 * Session ID persists indefinitely in localStorage.
 * 
 * @returns {string} UUID session identifier
 */
export const getOrCreateSessionId = (): string => {
  const SESSION_KEY = 'wwai_session_id';
  
  let sessionId = localStorage.getItem(SESSION_KEY);
  
  if (!sessionId) {
    // Generate new UUID v4
    sessionId = crypto.randomUUID();
    localStorage.setItem(SESSION_KEY, sessionId);
  }
  
  return sessionId;
};

/**
 * Track business info submission from create-site flow.
 * 
 * This fires when user clicks "Next" on the first step of onboarding,
 * regardless of whether they sign up or not.
 * 
 * @param data - Business information submitted by user
 */
export const trackOnboardingStep = async (data: {
  business_name?: string;
  google_maps_url?: string;
  yelp_url?: string;
  google_places_data?: any;
}): Promise<void> => {
  const sessionId = getOrCreateSessionId();
  
  try {
    await fetch(getBackendUrl('/api/product-analytics/onboarding-event'), {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        session_id: sessionId,
        business_name: data.business_name || null,
        google_maps_url: data.google_maps_url || null,
        yelp_url: data.yelp_url || null,
        google_places_data: data.google_places_data || null,
      }),
    });
    
    // Success - event tracked
    console.debug('✅ Onboarding event tracked', { sessionId });
  } catch (error) {
    // Don't block user flow if analytics fails
    console.error('⚠️ Failed to track onboarding event:', error);
  }
};

/**
 * Clear session ID (useful after successful business creation).
 * Call this to start fresh session for next business creation.
 */
export const clearAnalyticsSession = (): void => {
  localStorage.removeItem('wwai_session_id');
};

File: app/shared/controllers/analytics/product_analytics.py
"""
Product Analytics Controller

Tracks user behavior within the Wordsworth AI platform.
Unauthenticated endpoints for tracking anonymous user actions.
"""

from fastapi import APIRouter
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
import logging

from app.shared.services.analytics.product_analytics_service import ProductAnalyticsService

logger = logging.getLogger(__name__)

# Create router
product_analytics_router = APIRouter(
    prefix="/api/product-analytics",
    tags=["product-analytics"]
)


class OnboardingEventCreate(BaseModel):
    """Schema for onboarding event creation"""
    session_id: str = Field(..., description="Frontend-generated session ID from localStorage")
    business_name: Optional[str] = Field(None, description="Business name entered by user")
    google_maps_url: Optional[str] = Field(None, description="Google Maps URL entered by user")
    yelp_url: Optional[str] = Field(None, description="Yelp URL entered by user")
    google_places_data: Optional[Dict[str, Any]] = Field(None, description="Full Google Places API response")


@product_analytics_router.post("/onboarding-event")
async def track_onboarding_event(data: OnboardingEventCreate) -> Dict[str, Any]:
    """
    Track business info submission from create-site flow.
    
    NO AUTHENTICATION REQUIRED - tracks anonymous user behavior.
    
    Event is created every time user clicks "Next" on business info form,
    regardless of whether they sign up or not.
    """
    try:
        service = ProductAnalyticsService()
        event = await service.create_onboarding_event(
            session_id=data.session_id,
            business_name=data.business_name,
            google_maps_url=data.google_maps_url,
            yelp_url=data.yelp_url,
            google_places_data=data.google_places_data
        )
        
        return {
            "success": True,
            "event_id": str(event["_id"]),
            "message": "Event tracked successfully"
        }
    except Exception as e:
        logger.error(f"Failed to track onboarding event: {str(e)}", exc_info=True)
        # Don't fail the user's flow - return success even if tracking fails
        return {
            "success": False,
            "message": "Event tracking failed but user flow continues"
        }